source/application/SGMTests/SGMTestsWindow.cpp

```diff
@@ -40,7 +43,52 @@ void SGMTestsWindow::onShowMonoViewClicked()
                                                                   358.199551,
                                                                   this);
 
-    SGMMonochromatorInfoTestView* testView = new SGMMonochromatorInfoTestView(testMonoInfo);
-    testView->show();
+    monoInfoTestView_ = new SGMMonochromatorInfoTestView(testMonoInfo);
+    monoInfoTestView_->show();
+}
+
+void SGMTestsWindow::onShowTrajectoryViewClicked()
+{
+    if(energyTestView_ != 0) {
```

Comment:
I know this is just in your tests, but would it make more sense to do the inverse of this?  If you don't have one, then make one, but every subsequent call you just bring up the one that has been built already?